packages/beacon-node/src/api/impl/beacon/state/index.ts

```diff
@@ -23,7 +29,13 @@ export function getBeaconStateApi({
   async function getState(
     stateId: routes.beacon.StateId
   ): Promise<{state: BeaconStateAllForks; executionOptimistic: boolean; finalized: boolean}> {
-    return getStateResponse(chain, stateId);
+    const {state, executionOptimistic, finalized} = await getStateResponseWithRegen(chain, stateId);
+
+    return {
+      state: state instanceof Uint8Array ? loadState(config, chain.getHeadState(), state).state : state,
```

Comment:
do we wanna clone state here? I think the reason why I did it in https://github.com/ChainSafe/lodestar/pull/7130 was due to creating a `CachedBeaconState` from it (based on https://github.com/ChainSafe/lodestar/pull/7130/commits/35fa8e568a750887e93f856e604235d30ba2f34d)
```suggestion
      state: state instanceof Uint8Array ? loadState(config, chain.getHeadState(), state).state : state.clone(),
```

Reply:
shouldn't need it here, API should only need read only access to state